### ISSUE TITLE
Library does not specify Requests as a dependency

### ISSUE THREAD

**Issue by Madrox** [googleapis/google-auth-library-python#286]
As seen here, this lib requires Requests to be installed for some imports: https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/master/google/auth/transport/requests.py#L27

However, it is not specified as a dependency or extra in setup.py and there are no comments of tested versions of Requests. I imagine it isn't specified as a dependency because this is viewed as an optional module, but it should at least be included in extras. I mention this because I'm debugging an issue in my project that seems tied to the kwargs this library is attempting to send to Requests at https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/master/google/auth/transport/requests.py#L201. The only recourse I have is basically trial and error. I'm currenlty using Requests 2.19.1. 

I don't think it's the responsibility of this lib to handle this error, but it seems like it should at least specify which versions of Requests it has tested for support.

**Comment by tseaver** [comment]
I'm not sure how you could provoke an error from `google.auth.transport.requests`, as it re-raises the `ImportError` from trying to import `requests` at the [top of the module](https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/62cfc6dbfaad619a53e4fccaee203fa7516bc9d1/google/auth/transport/requests.py#L24).

I do agree that adding an "extra" for requests would be a useful signal to the user.

**Comment by Madrox** [comment]
@tseaver I have Requests 2.19.1 installed for other uses, so it's not raising an ImportError. However, I believe something is being passed to Requests as a kwarg that the version I'm using does not like. I'm getting a exception in Requests, not this lib. The exact line raising the exception is https://github.com/requests/requests/blob/v2.19.1/requests/sessions.py#L622 with `ConnectionError: ('Connection aborted.', error(22, 'Invalid argument'))`

**Comment by Madrox** [comment]
TL;DR....Requests 2.19.1 doesn't seem fully compatible with this lib. It will occasionally throw an except :)

**Comment by theacodes** [comment]
We should be completely compatible with everything in the 2.x series. Can you give us a full stacktrace?

**Comment by Madrox** [comment]
```Traceback (most recent call last):
2018-08-27 15:20:29.995 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/django/core/handlers/exception.py", line 41, in inner
2018-08-27 15:20:29.995 PDT
    response = get_response(request)
2018-08-27 15:20:29.995 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/django/core/handlers/base.py", line 187, in _get_response
2018-08-27 15:20:29.995 PDT
    response = self.process_exception_by_middleware(e, request)
2018-08-27 15:20:29.995 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/django/core/handlers/base.py", line 185, in _get_response
2018-08-27 15:20:29.995 PDT
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
2018-08-27 15:20:29.995 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/django/views/decorators/csrf.py", line 58, in wrapped_view
2018-08-27 15:20:29.995 PDT
    return view_func(*args, **kwargs)
2018-08-27 15:20:29.995 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/django/views/generic/base.py", line 68, in view
2018-08-27 15:20:29.995 PDT
    return self.dispatch(request, *args, **kwargs)
2018-08-27 15:20:29.998 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/rest_framework/views.py", line 483, in dispatch
2018-08-27 15:20:29.998 PDT
    response = self.handle_exception(exc)
2018-08-27 15:20:29.998 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/rest_framework/views.py", line 443, in handle_exception
2018-08-27 15:20:29.999 PDT
    self.raise_uncaught_exception(exc)
2018-08-27 15:20:29.999 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/rest_framework/views.py", line 480, in dispatch
2018-08-27 15:20:30.010 PDT
    response = handler(request, *args, **kwargs)
2018-08-27 15:20:30.011 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/rest_framework/generics.py", line 254, in get
2018-08-27 15:20:30.091 PDT
    return self.retrieve(request, *args, **kwargs)
2018-08-27 15:20:30.091 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/rest_framework/mixins.py", line 58, in retrieve
2018-08-27 15:20:30.091 PDT
    return Response(serializer.data)
2018-08-27 15:20:30.092 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/rest_framework/serializers.py", line 560, in data
2018-08-27 15:20:30.092 PDT
    ret = super(Serializer, self).data
2018-08-27 15:20:30.092 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/rest_framework/serializers.py", line 262, in data
2018-08-27 15:20:30.092 PDT
    self._data = self.to_representation(self.instance)
2018-08-27 15:20:30.092 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/rest_framework/serializers.py", line 527, in to_representation
2018-08-27 15:20:30.092 PDT
    ret[field.field_name] = field.to_representation(attribute)
2018-08-27 15:20:30.092 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/rest_framework/fields.py", line 1528, in to_representation
2018-08-27 15:20:30.092 PDT
    if not getattr(value, 'url', None):
2018-08-27 15:20:30.092 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/django/db/models/fields/files.py", line 70, in url
2018-08-27 15:20:30.092 PDT
    return self.storage.url(self.name)
2018-08-27 15:20:30.092 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/storages/backends/gcloud.py", line 246, in url
2018-08-27 15:20:30.093 PDT
    blob = self._get_blob(self._encode_name(name))
2018-08-27 15:20:30.093 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/storages/backends/gcloud.py", line 210, in _get_blob
2018-08-27 15:20:30.093 PDT
    blob = self.bucket.get_blob(name)
2018-08-27 15:20:30.093 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/google/cloud/storage/bucket.py", line 401, in get_blob
2018-08-27 15:20:30.093 PDT
    _target_object=blob,
2018-08-27 15:20:30.093 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/google/cloud/_http.py", line 290, in api_request
2018-08-27 15:20:30.093 PDT
    headers=headers, target_object=_target_object)
2018-08-27 15:20:30.093 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/google/cloud/_http.py", line 183, in _make_request
2018-08-27 15:20:30.093 PDT
    return self._do_request(method, url, headers, data, target_object)
2018-08-27 15:20:30.093 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/google/cloud/_http.py", line 212, in _do_request
2018-08-27 15:20:30.093 PDT
    url=url, method=method, headers=headers, data=data)
2018-08-27 15:20:30.093 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/google/auth/transport/requests.py", line 201, in request
2018-08-27 15:20:30.098 PDT
    method, url, data=data, headers=request_headers, **kwargs)
2018-08-27 15:20:30.098 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/requests/sessions.py", line 512, in request
2018-08-27 15:20:30.098 PDT
    resp = self.send(prep, **send_kwargs)
2018-08-27 15:20:30.098 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/requests/sessions.py", line 622, in send
2018-08-27 15:20:30.098 PDT
    r = adapter.send(request, **kwargs)
2018-08-27 15:20:30.098 PDT
  File "/base/data/home/apps/s~leaguerank-api/20180826t235731.412136437085760579/lib/requests/adapters.py", line 495, in send
2018-08-27 15:20:30.100 PDT
    raise ConnectionError(err, request=request)
2018-08-27 15:20:30.100 PDT
ConnectionError: ('Connection aborted.', error(22, 'Invalid argument'))```

**Comment by Madrox** [comment]
I haven't gotten far in debugging this since the full trace involves two frameworks and three libraries. It's entirely possible this is because of something wrong DRF, Django, or the GCS storage lib are doing upstream. If this lib is fully tested on 2.X I'm probably barking up the wrong tree with this issue.

**Comment by tseaver** [comment]
It might help us debug by eye if you could set a breakpoint inside `google/auth/transport/requests.py` (via `import pdb; pdb.set_trace()`).  Put that line just before line 201 and show the values of the variables passed to the `super` method call, e.g.:

```python
(pdb) pp method
...
(pdb) pp url
...
(pdb) pp data
...
(pdb) pp request_headers
...
(pdb) pp kwargs
```

**Comment by tseaver** [comment]
Please feel free to follow up / reopen if you can provide more information.
